moz_security/sandboxes/heka/output/moz_security_metrics.lua

```diff
@@ -0,0 +1,120 @@
+-- This Source Code Form is subject to the terms of the Mozilla Public
+-- License, v. 2.0. If a copy of the MPL was not distributed with this
+-- file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+--[[
+# Mozilla Security Metrics Output Plugin
+
+This output module collects metrics generated using the secmetrics Heka module,
+aggregates them, and outputs them based on the configured ticker interval.
+
+If using email based output via the email_recipient option, ensure you have
+a valid email alert configuration. The email functionality in this sandbox makes
+use of the existing code present for email based alerting.
+
+## Sample Configuration
+```lua
+filename = "moz_security_metrics.lua"
+message_matcher = "Type == 'secmetrics'"
+ticker_interval = 28800
+
+-- debug = false -- optional, if set emit aggregated stats in debug log
+
+-- email_recipient = "riker@mozilla.com" -- optional, if set emit metrics to specified recipient
+-- alert = { modules = { email = { ... } } } -- optional, required if email_recipient is set
+```
+--]]
+
+require "cjson"
+
+local debug         = read_config("debug")
+local recipient     = read_config("email_recipient")
+local cephost       = read_config("Hostname")
+
+_PRESERVATION_VERSION = read_config("preservation_version") or 0
+
+z = {} -- global, reload on startup
+
+local malert
+if recipient then
+    malert = require "alert.email"
+end
+
+local function fold()
+    local ret = {}
+    for k,v in pairs(z) do
+        ret[k] = {}
+        for x,y in pairs(v.accumulators) do
+            ret[k][x] = y
+        end
+        for x,y in pairs(v.uniqitems) do
+            local cnt = 0
+            for l,w in pairs(y) do
+                cnt = cnt + 1
+            end
+            assert(not ret[k][x], "collision in metric name for " .. k)
```

Comment:
This will terminate the plugin and end all aggregation, is that the desired behavior?

Reply:
Yeah I think this is ok